src/main/java/games/strategy/engine/lobby/server/login/LobbyLoginValidator.java

```diff
@@ -26,49 +28,50 @@
 import games.strategy.util.Version;
 
 public class LobbyLoginValidator implements ILoginValidator {
-  static final String THATS_NOT_A_NICE_NAME = "That's not a nice name.";
-  private static final String YOU_HAVE_BEEN_BANNED = "You have been banned from the TripleA lobby.";
-  private static final String USERNAME_HAS_BEEN_BANNED = "This username is banned, please create a new one.";
-  private static final String UNABLE_TO_OBTAIN_MAC = "Unable to obtain mac address.";
-  private static final String INVALID_MAC = "Invalid mac address.";
-  private static final Logger s_logger = Logger.getLogger(LobbyLoginValidator.class.getName());
   public static final String LOBBY_VERSION = "LOBBY_VERSION";
   public static final String REGISTER_NEW_USER_KEY = "REGISTER_USER";
   public static final String ANONYMOUS_LOGIN = "ANONYMOUS_LOGIN";
   public static final String LOBBY_WATCHER_LOGIN = "LOBBY_WATCHER_LOGIN";
   public static final String HASHED_PASSWORD_KEY = "HASHEDPWD";
   public static final String EMAIL_KEY = "EMAIL";
   public static final String SALT_KEY = "SALT";
-
-  public LobbyLoginValidator() {}
+  static final String THATS_NOT_A_NICE_NAME = "That's not a nice name.";
+  private static final String YOU_HAVE_BEEN_BANNED = "You have been banned from the TripleA lobby.";
+  private static final String USERNAME_HAS_BEEN_BANNED = "This username is banned, please create a new one.";
+  private static final String UNABLE_TO_OBTAIN_MAC = "Unable to obtain mac address.";
+  private static final String INVALID_MAC = "Invalid mac address.";
+  private static final Logger logger = Logger.getLogger(LobbyLoginValidator.class.getName());
 
   @Override
   public Map<String, String> getChallengeProperties(final String userName, final SocketAddress remoteAddress) {
     // we need to give the user the salt key for the username
     final Map<String, String> rVal = new HashMap<>();
-    final HashedPassword password = new DbUserController().getPassword(userName);
+    final HashedPassword password = new DbUserController().getLegacyPassword(userName);
     if (password != null && Strings.emptyToNull(password.value) != null) {
       rVal.put(SALT_KEY, MD5Crypt.getSalt(MD5Crypt.MAGIC, password.value));
     }
+    RsaAuthenticator.appendPublicKey(rVal);
```

Comment:
bit low level, from a reader point of view, void methods are always a bit difficult since it is unclear exactly what is happening. This is of a bit more marginal value, but I recommend to have `RsaAuthenitcator.appendPublicKey` to instead be updated to return a `Pair`. It should look nicer in test I suspect too, the verification would more direct.